SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -27,58 +27,175 @@ This SWIP is part of a series of SWIPs (but can be implemented on its own). To s
 
 1. Currency to use
 2. Price oracle to use
-3. Payment method to use
+3. Payment provider to use
+
+We propose a standardized way to communicate these preferences and an algorithm to resolve any two preferences lists.
+
+For any dimension, nodes list the relative importance of them. Consequently, nodes specify their accepted options within each dimension. Importance is communicated through Swarm by any positive integer
+
+The following is an example of the proposed configuration:
+
+```yaml
+# The relative importance of each dimension for the node
+dimensions:
+	currency: 70
+	provider: 20
+	oracle: 10
+
+# Supported payment options by the node.
+# A payment module is defined by a currency, provider and oracle. Weights are the associated preferences.
+currencies:
+	rif:
+		weight: 36
+		providers:
+			lumino:
+				weigth: 15
+			raiden:
+				weight: 45
+		oracles:
+			rifOracleA:
+				weigth: 10
+			rifOracleB:
+				weigth: 70
+			rifOracleC:
+				weigth: 60
+	xdai:
+		weight: 9
+		providers:
+			Lumino: 
+				weigth: 10
+		oracles:
+			xDaiOracle: 
+				weight: 20
+	dai:
+		weight: 5
+		providers:
+			raiden:
+				weight: 80
+		oracles:
+			daiOracleA: 
+				weight: 30
+			daiOracleB:
+				weight: 20
+	eth:
+		weight: 40
+		providers:
+			swap:
+				weight: 35
+		oracles:
+			ethOracleA:
+				weight: 90
+			ethOracleB:
+				weight: 10
+```
+
+This configuration is used by the payment method selection algorithm in the following way: 
+
+1. Based on the node configuration generate a set of triplets in the form ```[currency, provider, oracle]```. As an example let's take from the previous configuration the ```rif``` entry. From that entry we can build the triplets `[rif, lumino, rifOracleA]`, `[rif, lumino, rifOracleB]`, `[rif, lumino, rifOracleC]`,`[rif, raiden, rifOracleA]`, `[rif, raiden, rifOracleB]`, `[rif, raiden, rifOracleC]`, but triplets are generated for the other currencies as well. 
+2. Exchange the generated set of triplets with the peer B.
+3. Keep the set of triplets which are common (intersection between tripletsA and tripletsB). If no common set exist, choose the fallback option for each dimension, as shown in the following table:
+
+	| Dimension  | Fallback option |
+	| ---------- | --------------- |
+	| Currency to use | Wei (Ether)                    |
+	| Price oracle    | HonMon oracle <SWIP reference> |
+	| Payment method  | Chequebook contract            |
+
+4. Compute the weighted preference of each remaining triplet from node A and node B as:
+
+	```golang
+	weightedPreference := normalized(dimensions.currency) * normalized(triplet[n].weight) + 
+		+ normalized(dimensions.provider) * normalized(triplet[n].provider.weight) + 
+		+ normalized(dimensions.oracle) * normalized(triplet[n].oracle.weight)
+	```
+
+	Since weights and dimensions do not need to add up to a particular value (e.g. 100) they need to be normalized.
```

Comment:
out of curiosity: why did you choose not to make users supply normalized preferences? was it usability?

Reply:
Yes. Imagine that the user manually normalized the weights to 100 and later it wants to introduce a new weight, either because he added a new currency or a new oracle. In this case the user should adjust all the weights to normalize them to 100. Another benefit of doing the normalization in the code is that we are not affected by an error from the user if the weights are misconfigured.

Reply:
cleared up, thanks.